pallets/asset-manager/src/migrations.rs

```diff
@@ -0,0 +1,416 @@
+							GeneralIndex(index),
+						),
+					};
+					let new_asset_type: T::AssetType = new_location.into();
+					// Insert new asset type previous asset type
+					AssetIdType::<T>::insert(&asset_id, &new_asset_type);
+
+					// This is checked in case AssetManagerPopulateAssetTypeIdStorage runs first
+					if AssetTypeId::<T>::get(&asset_type) == Some(asset_id) {
+						// We need to update AssetTypeId too
+						AssetTypeId::<T>::remove(&asset_type);
+						AssetTypeId::<T>::insert(&new_asset_type, asset_id);
+
+						// Update weight due to this branch
+						used_weight = used_weight.saturating_add(2 * db_weights.write);
+					}
+
+					// Update used weight
+					used_weight = used_weight.saturating_add(db_weights.write + db_weights.read);
+				}
+				_ => continue,
+			}
+		}
+
+		log::info!(target: "ChangeStateminePrefixes", "almost done");
+
+		// Return the weight used.
+		used_weight
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<(), &'static str> {
+		use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+		let pallet_prefix: &[u8] = b"AssetManager";
+		let storage_item_prefix: &[u8] = b"AssetIdType";
+
+		// We want to test that:
+		// If there exists an assetType matching the Statemine, it gets overwritten
+
+		// Check number of entries, and set it aside in temp storage
+		let stored_data: Vec<_> = storage_key_iter::<T::AssetId, T::AssetType, Blake2_128Concat>(
+			pallet_prefix,
+			storage_item_prefix,
+		)
+		.collect();
+
+		let statemine_para_id = StatemineParaIdInfo::get();
+
+		let mut found = false;
+
+		for (asset_id, asset_type) in stored_data {
+			let location: Option<MultiLocation> = asset_type.clone().into();
+			match location {
+				Some(MultiLocation {
+					parents: 1,
+					interior: X2(Parachain(para_id), GeneralIndex(_)),
+				}) if para_id == statemine_para_id => {
+					// We are going to note that we found at least one entry matching
+					found = true;
+					// And we are going to record its data
+					Self::set_temp_storage((asset_id, asset_type), "example_pair");
+					break;
+				}
+				_ => continue,
+			}
+		}
+		Self::set_temp_storage(found, "matching_type_found");
+
+		Ok(())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade() -> Result<(), &'static str> {
+		use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+		// Check if we found a matching type
+		let found: bool = Self::get_temp_storage("matching_type_found")
+			.expect("We stored a matching_type_found and should be here; qed");
+
+		let statemine_para_id = StatemineParaIdInfo::get();
+		let statemine_assets_pallet = StatemineAssetsInstanceInfo::get();
+
+		// Check that our example pair suffered the correct migration
+		if found {
+			let (asset_id, asset_type): (T::AssetId, T::AssetType) =
+				Self::get_temp_storage("example_pair").expect("qed");
+			let location: Option<MultiLocation> = asset_type.into();
+
+			match location {
+				Some(MultiLocation {
```

Comment:
You could also throw in an assert here that makes sure the `From` impl can be used as expected with respect to this conversion

Reply:
Not entirely sure about what you mean by this. Can you elaborate more?